ssz/sedes/base.py

```diff
@@ -104,77 +85,109 @@ def __init__(self, size: int):
     #
     # Size
     #
-    @property
-    def is_static_sized(self):
-        return True
+    is_fixed_sized = True
 
-    def get_static_size(self):
+    def get_fixed_size(self):
         return self.size
 
     #
-    # Serialization
+    # Tree hashing
     #
-    def serialize(self, value: TSerializable) -> bytes:
-        serialized_content = self.serialize_content(value)
-        if len(serialized_content) != self.size:
-            raise SerializationError(f"Cannot serialize {value} in {self.size} bytes")
-        return serialized_content
+    def hash_tree_root(self, value: TSerializable) -> bytes:
+        serialized_value = self.serialize(value)
+        return merkleize(pack((serialized_value,)))
 
-    @abstractmethod
-    def serialize_content(self, value: TSerializable) -> bytes:
-        pass
 
-    #
-    # Deserialization
-    #
-    def deserialize_segment(self, data: bytes, start_index: int) -> Tuple[TDeserialized, int]:
-        content, continuation_index = self.consume_bytes(data, start_index, self.size)
-        return self.deserialize_content(content), continuation_index
+class BaseCompositeSedes(BaseSedes[TSerializable, TDeserialized]):
+    pass
+
+
+def _compute_section_0_size(element_sedes: Iterable[TSedes]) -> int:
```

Comment:
IMO `section_0` and `section_1` are confusing names. `fixed_parts` and `variable_parts` from the current https://github.com/ethereum/eth2.0-specs/pull/787 PR are better.